src/components/data-exploration/hierarchical-tree/HierarchicalTree.jsx

```diff
@@ -50,24 +50,27 @@ const HierarchicalTree = (props) => {
     // If rootNode, ignore
     if (dragNode.rootNode) return;
 
+    if (dropPosition === -1) return;
+
     // pos is a string e.g.: 0-0-1, each number is a position in a tree level
     const posFromArray = dragNode.pos.split('-');
     const posToArray = node.pos.split('-');
 
-    const fromPosition = parseInt(posFromArray[2], 10);
+    const sameLevel = (posFromArray.length === posToArray.length);
 
     // If not in the same cellClass, ignore
     if (!_.isEqual(posFromArray[1], posToArray[1])) return;
 
-    // If was dropped in same place, ignore
-    if (fromPosition === dropPosition) return;
+    const numberOfClusters = treeData[posFromArray[1]].children.length;
 
-    // If not dropped in gap, ignore
-    // (only allow dropToGap when the destination node is rootNode
-    // because it can have children nodes)
-    if (!dropToGap && !node.rootNode) return;
+    const fromPosition = parseInt(posFromArray[2], 10);
+    const toPosition = (!sameLevel ? dropToGap ? numberOfClusters : 0 : dropPosition);
+
+    // If was dropped in same place, ignore
+    if (fromPosition === toPosition) return;
 
-    const newPosition = dropPosition - (fromPosition < dropPosition ? 1 : 0);
+    // if dropping below the initial position subtract 1, if dropping to secondary position add 1
```

Comment:
what do you mean by secondary position?

Reply:
When dropping between clusters there are two possible drop positions. The secondary position is the one to the right -> with `DropToGap: false` and `dragOver: true`. It has `DropPosition` offset by 1 (due to antd).